### PR TITLE
Homework 10 pattern, IO

### PR DESCRIPTION
**Pull request by goharminasyan**
1. validated DTO through chain of processors - chain of responsibility
2. IO

### CODE REVIEW

File: src/io/FileUtil.java [hunk 1]
```diff
@@ -1,52 +1,95 @@
...
+                outputStream.write(bytes);
+                if (i + 1 > 98999999) {
+                    outputStream.close();
+                }
+            }
+        } catch (FileNotFoundException e) {
+            System.out.printf("File %s not found", filePhoneNumbers);
+        } catch (IOException e) {
+            System.out.println("Error during write");
+        }
```
Comment (Karlen999) [inline src/io/FileUtil.java:54]: Use try-with-resources or close this **"FileOutputStream"**  in a "finally" clause.

File: src/io/FileUtil.java [hunk 2]
```diff
@@ -1,52 +1,95 @@
...
-    public static void serialize(User user, String filePath){
+    public static void serialize(User user, String filePath) {
 
+        try {
+            ObjectOutputStream objectOut = new ObjectOutputStream(new FileOutputStream(filePath));
+            objectOut.writeObject(user.toString());
+            objectOut.close();
+        } catch (Exception ex) {
+            ex.printStackTrace();
+        }
```
Comment (Karlen999) [inline src/io/FileUtil.java:72]: Use try-with-resources or close this **"ObjectOutputStream"**  in a "finally" clause.

File: src/io/FileUtil.java [hunk 3]
```diff
@@ -1,52 +1,95 @@
...
      * @return
      */
-    public static User deserialize(String filePath){
+    public static Object deserialize(String filePath) {
+         Object o = null;
+        try {
+            FileInputStream fileInputStream = new FileInputStream(filePath);
+            ObjectInputStream objectInputStream = new ObjectInputStream(fileInputStream);
+            o = objectInputStream.readObject();
+            objectInputStream.close();
```
Comment (Karlen999) [inline src/io/FileUtil.java:87]: Use try-with-resources or close these **"FileInputStream"**  and **"ObjectInputStream"** in a "finally" clause.

File: src/io/FileUtilTest.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        address.setCountry("Armenia");
+        address.setCity("Gyumri");
+        address.setStreet("16 street");
+        address.setZipCode("3101");
+        address.setLine1("12/3");
+        user.setAddress(address);
+
+        FileUtil.serialize(user, "C:\\Users\\Admin\\trainings\\epam-gyumri-trainings\\src\\io\\dir\\InfoAboutUser.txt");
+
+        System.out.println(FileUtil.deserialize("C:\\Users\\Admin\\trainings\\epam-gyumri-trainings\\src\\io\\dir\\InfoAboutUser.txt"));
```
Comment (Karlen999) [inline src/io/FileUtilTest.java:30]: You can declare final variables for _"pathname"_  and _"filePath"_ and call them

File: src/io/User.java [hunk 5]
```diff
@@ -35,7 +46,10 @@ public void setPassword(String password) {
     }
 
     public String getBankCardNumber() {
-        return bankCardNumber;
+        byte [] cardNumber = bankCardNumber.getBytes();
+        Base64.getEncoder().encode(cardNumber);
+        String s = cardNumber.toString();
```
Comment (Karlen999) [inline src/io/User.java:51]: Use "Arrays.toString()" instead.

Comment (goharminasyan) [reply src/io/User.java:51]: But Arrays.toString()  returns us a mass of bytes, but we need to show the data through encrypted Base64․ It is not so?

**Comment by Karlen999** [comment]
reviewed

**Pull request closed without merging**